shell/platform/darwin/ios/ios_surface.mm

```diff
@@ -90,15 +90,19 @@ bool IsIosEmbeddedViewsPreviewEnabled() {
   platform_views_controller_->CancelFrame();
   // Committing the current transaction as |BeginFrame| will create a nested
   // CATransaction otherwise.
-  [CATransaction commit];
+  if ([NSThread currentThread] == [NSThread mainThread]) {
+    [CATransaction commit];
+  }
 }
 
 // |ExternalViewEmbedder|
 void IOSSurface::BeginFrame(SkISize frame_size, GrContext* context, double device_pixel_ratio) {
   TRACE_EVENT0("flutter", "IOSSurface::BeginFrame");
   FML_CHECK(platform_views_controller_ != nullptr);
   platform_views_controller_->SetFrameSize(frame_size);
-  [CATransaction begin];
+  if ([NSThread currentThread] == [NSThread mainThread]) {
```

Comment:
qq: in which other thread can this code run?

Reply:
it used to also run on GPU thread when there is no platform view.

Reply:
sg. what about adding a comment here and else where about this behavior?

Reply:
can do!